sources/packages/web/src/views/student/application-offering-change-request/ApplicationOfferingChangeFormView.vue

```diff
@@ -0,0 +1,187 @@
+</template>
+
+<script lang="ts">
+import { useRouter, RouteLocationRaw } from "vue-router";
+import { defineComponent, ref, computed, onMounted, watchEffect } from "vue";
+import { StudentRoutesConst } from "@/constants/routes/RouteConstants";
+import ApproveApplicationOfferingChangeRequestModal from "@/components/students/modals/ApproveApplicationOfferingChangeRequestModal.vue";
+import DeclineApplicationOfferingChangeRequestModal from "@/components/students/modals/DeclineApplicationOfferingChangeRequestModal.vue";
+import { ModalDialog, useSnackBar } from "@/composables";
+import { ApplicationOfferingChangeRequestService } from "@/services/ApplicationOfferingChangeRequestService";
+import {
+  ApplicationOfferingChangeRequestStatusAPIOutDTO,
+  StudentApplicationOfferingChangeRequestAPIInDTO,
+} from "@/services/http/dto";
+import { ApplicationOfferingChangeRequestStatus } from "@/types";
+
+export default defineComponent({
+  components: {
+    ApproveApplicationOfferingChangeRequestModal,
+    DeclineApplicationOfferingChangeRequestModal,
+  },
+  props: {
+    applicationOfferingChangeRequestId: {
+      type: Number,
+      required: true,
+    },
+    applicationId: {
+      type: Number,
+      required: false,
+    },
+  },
+  setup(props) {
+    const snackBar = useSnackBar();
+    const router = useRouter();
+    const applicationOfferingChangeRequestStatus =
+      ref<ApplicationOfferingChangeRequestStatusAPIOutDTO>();
+    const approveApplicationOfferingChangeRequestModal = ref(
+      {} as ModalDialog<StudentApplicationOfferingChangeRequestAPIInDTO>,
+    );
+    const declineApplicationOfferingChangeRequestModal = ref(
+      {} as ModalDialog<StudentApplicationOfferingChangeRequestAPIInDTO>,
+    );
+    const getLabel = () => {
+      return applicationOfferingChangeRequestStatus.value?.status ===
+        ApplicationOfferingChangeRequestStatus.Approved
+        ? "Previous application details"
+        : "Active application details";
+    };
+    const goBackRouteParams = computed(
+      () =>
+        ({
+          name: StudentRoutesConst.STUDENT_APPLICATION_DETAILS,
+          params: { id: props.applicationId },
+        } as RouteLocationRaw),
+    );
+    const items = ref([
+      {
+        label: "Requested change",
+        command: () => ({
+          name: StudentRoutesConst.STUDENT_REQUESTED_APPLICATION_OFFERING_CHANGE,
+          params: {
+            applicationOfferingChangeRequestId:
+              props.applicationOfferingChangeRequestId,
+            applicationId: props.applicationId,
+          },
+        }),
+      },
+      {
+        label: getLabel(),
+        command: () => ({
+          name: StudentRoutesConst.STUDENT_ACTIVE_APPLICATION_DETAILS,
+          params: {
+            applicationOfferingChangeRequestId:
+              props.applicationOfferingChangeRequestId,
+            applicationId: props.applicationId,
+          },
+        }),
+      },
+    ]);
+    watchEffect(() => {
```

Comment:
https://markus.oberlehner.net/blog/watch-vs-watcheffect-when-to-use-what-with-vue/,https://stackoverflow.com/questions/60510895/vue-3-composition-api-watcheffect-vs-watch